app/controllers/api/v1/targets_controller.rb

```diff
@@ -0,0 +1,42 @@
+module Api
+  module V1
+    class TargetsController < ApplicationController
+      before_action :authenticate_user!
+      before_action :set_target, only: [:show]
+
+      def index
+        @targets = Target.all
+      end
+
+      def show; end
+
+      def create
+        new_target
+        @target.save ? render_create_success : render_create_error
```

Comment:
By using `create!()` you can change this line to `render :show`. If any errors occur, Rails would throw an exception (like `ActiveRecord::RecordNotFound`, `ActiveRecord::RecordInvalid` or `ActionController::ParameterMissing`). I would suggest catching these exceptions on the application controller and returning the correct message and status code.

Reply:
I do this because this way i can return the error messages by field individually. I think this way is better for front end to show specifics errors by field. Using what you suggest returns only one text with all the errors together.